services/slack_grid/extract.go

```diff
@@ -0,0 +1,201 @@
+package slack_grid
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const DefaultDirPath = "tmp/slack_grid"
+
+func (t *BulkTransformer) GetWorkingDir() (string, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return "", errors.Wrap(err, "error getting current working directory")
+	}
+	return dir, nil
+}
+
+func (t *BulkTransformer) readDir(dest string) ([]fs.DirEntry, error) {
+	files, err := os.ReadDir(dest)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("error reading directory %v", dest))
+	}
+	return files, nil
+}
+
+func (t *BulkTransformer) dirHasContent(dest string) (bool, error) {
```

Comment:
Need to add a check here to allow them to overwrite existing data. Right now it's quite hidden and enables errors